### DISCUSSION TITLE
How to use the Gpio for LED

### DISCUSSION THREAD

**Discussion by dernassesmart** [rednblkx/HomeKey-ESP32#61] [Q&A]
Hello and thank you very much for this excellent repository. It’s fun to follow your guide and build it. I noticed that there is an option for the LED in the configuration file. My question is whether I need to add a resistor in between or if I can simply use any LED. And what about colored LEDs?
Thank you in advance.

**Comment by rednblkx** [comment]
Hello!

I'm glad you like it.

Note that the `LED_PIN` in `config.h` is strictly used by the [HomeSpan](https://github.com/HomeSpan/HomeSpan) library as a status LED(see more in their [docs](https://github.com/HomeSpan/HomeSpan/blob/master/docs/UserGuide.md#homespan-device-states)). This has been renamed in the `dev` branch to better reflect its usage.

In the `dev` branch, among other things, now you have the option to assign a GPIO pin to be actioned on a successful/failed HomeKey Auth if that interests you. I'm still working on things so the wiki won't be updated yet until i've merged `dev` into `main`.

Following your question about adding a resistor, unless the board already includes a resistor of the right value on the pin, you should always make sure to add a resistor on the led's anode to not fry it.

Not sure if you are referring to an RGB led, but if you do, that requires 3 separate GPIO pins for each of the colors.

Let me know if you have any other questions.

**Reply by dernassesmart** [reply]
thank you very much for the detailed answer. i think the dev branch is what i am looking for!
